torchvision/transforms/functional_tensor.py

```diff
@@ -48,3 +48,25 @@ def crop(img, top, left, height, width):
         raise TypeError('tensor is not a torch image.')
 
     return img[..., top:top + height, left:left + width]
+
+def to_grayscale(img, num_output_channels = 3):
+    """Convert the given RGB Image Tensor to Grayscale.
+
+    Args
+        img (Tensor): Image to be converted to Grayscale in the form [C, H, W].
+        num_output_channels (int): denotes the number of channels to return after conversion
+    Returns:
+    Tensor: Grayscale image.
+    """
+    if not F._is_tensor_image(img):
+        raise TypeError('tensor is not a torch image.')
+
+    else:
```

Comment:
Can you remove this `else:` branch? It's not necessary because we raise an error in the other branch

Reply:
Also, can you add a check that the input image has `3` channels?